app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java

```diff
@@ -134,6 +146,69 @@ public void startDownload(@NonNull DownloadJob job) {
         }
     }
 
+    public void downloadBlobUri(DownloadJob job) {
+        if (!UrlUtils.isBlobUri(job.getUri()) || job.getInputStream() == null) {
+            Log.w(LOGTAG, "Failed to download Blob URI: " + job.getUri());
+            return;
+        }
+
+        final File dir = mContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS);
+        if (dir == null || (dir.exists() && !dir.isDirectory())) {
```

Comment:
Do we really need this? I mean, that directory is provided by Android so I doubt that it could happen that it does not exist or that is not a directory.

Reply:
`Context.getExternalFilesDir()` warns that it might return `null` in some cases, so I left the null-check in place.